### PR TITLE
Hide the Add Contact button if we've got an invalid contact

### LINKED ISSUE #160: When a new contact has no details at all filled in yet, it should not be possible to add another one
Right now you can just add empty contacts endlessly, which makes little sense.

When a new contact has no specific info filled in yet, clicking »Add new contact« should just focus the name field of the already added one.

### PR DESCRIPTION
**Pull request by skjnldsv**
This prevent from adding multiple empty contacts as long as you don't have filled the full name field (aka FN).
- Hides the new contact button when we have an invalid contact
- Resized the button height to match a contactList item height.
- A new contact now take the place of the button.

This fixes #160
screencast

### CODE REVIEW

File: js/components/contactList/contactList_controller.js [hunk 1]
```diff
@@ -145,6 +146,15 @@ angular.module('contactsApp')
 		}
 	});
 
+	// Watch if we have an invalid contact
+	$scope.$watch('ctrl.contactList[0].fullName()', function(fullName) {
```
Comment (DeepDiver1975) [inline js/components/contactList/contactList_controller.js:154]: Is it always the first contact in the list?
Looks strange to me

File: js/components/contactList/contactList_controller.js [hunk 2]
```diff
@@ -145,6 +146,15 @@ angular.module('contactsApp')
...
+		if(fullName === '') {
+			ctrl.invalid = true;
```
Comment (DeepDiver1975) [inline js/components/contactList/contactList_controller.js]: Can we a one liner ...

File: js/components/contactList/contactList_controller.js (hunk 1 above)
Comment (skjnldsv) [inline js/components/contactList/contactList_controller.js:154]: if the fn field is empty, and therefore the contact invalid, then yes, always the first one on the list.

File: js/components/contactList/contactList_controller.js (hunk 2 above)
Comment (skjnldsv) [inline js/components/contactList/contactList_controller.js]: Done

**Comment by jancborchardt** [comment]
Only disable the button instead of hiding it. That will prevent the list from making this strange jump.

**Comment by skjnldsv** [comment]
The button hiding is exactly to prevent the list from making the jump, it replaces the button with the same height and therefore does not move your whole list to the bottom. I though this was a good idea.

**Comment by jancborchardt** [comment]
Actually good point, yeah. Let's do it like that.

I have some ideas on how to make the »Add contact« function look more nicely in the list than just a button too.

**Comment by jancborchardt** [comment]
Very nice!

Two small enhancements:
- until a proper name is put in, show »New contact« in the first line of that newly created contact
- when clicking on another contact, and then back on that newly created contact (without any info put in yet), the full name field should automatically be focused

**Comment by skjnldsv** [comment]
Good ideas!

**Comment by skjnldsv** [comment]
New contact is now showed. For the focus I'm having trouble, it does not seems to work. I tried adding 

```
if(ctrl.contact.fullName() === '') {
    $('#details-fullName').focus();
}
```

On contactsDetails_controller.js in the `ContactService.getById(uid).then(function(contact) {` line.
Doesn't seems to work.

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**